_posts/2021-11-19-connecting-java-high-level-rest-client-with-opensearch-over-https.markdown

```diff
@@ -0,0 +1,150 @@
+
+Use the Java `keytool` to create a custom truststore and import the root authority certificate. The `keytool` does not understand the `.pem` format, so we’ll have to first convert the root authority certificate to `.der` format using `openssl` cryptography library and then add it to the custom truststore using Java `keytool`. Most Linux distributions already come with `openssl` installed.
+
+_STEP 1:_ Convert the root authority certificates from .pem to .der format.
+
+```
+openssl x509 -in opensearch-1.1.0/config/root-ca.pem -inform pem -out root-ca.der --outform der
+```
+
+_STEP 2:_ Create a custom truststore and add the `root-ca.der` certs. Later, we will set this custom truststore in the client code to override the default truststore.
+
+```
+keytool -import root-ca.der -alias opensearch -keystore myTrustStore
+```
+
+Confirm the action was successful by listing certs in truststore. The grep should be able to find opensearch alias if the certs were added successfully.
+
+```
+keytool -keystore myTrustStore -storepass changeit -list | grep opensearch
+```
+
+_STEP 3:_ Next, set the truststore properties in the client code to point to the custom truststore. 
+
+```
+//Point to keystore with appropriate certificates for security.
+System.setProperty("javax.net.ssl.trustStore", "/full/path/to/myCustomTrustStore");
+System.setProperty("javax.net.ssl.trustStorePassword", "password-for-myCustomTrustStore");
+```
+
+
+_STEP 4:_ Now create an instance of client and connect with OpenSearch over HTTPS. 
+
+```
+// Create the client.
+RestClientBuilder builder = RestClient.builder(new HttpHost("localhost", 9200, "https"))
+        .setHttpClientConfigCallback(new RestClientBuilder.HttpClientConfigCallback() {
+            @Override
+            public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpClientBuilder) {
+                return httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+            }
+        });
+RestHighLevelClient client = new RestHighLevelClient(builder);
+
+// Create an index with custom settings.
+CreateIndexRequest createIndexRequest = new CreateIndexRequest("twitter");
+createIndexRequest.settings(Settings.builder() //Specify in the settings how many shards you want in the index.
+    .put("index.number_of_shards", 4)
+    .put("index.number_of_replicas", 1)
+    );
+CreateIndexResponse createIndexResponse = client.indices().create(createIndexRequest, RequestOptions.DEFAULT);
+```
+
+You can also checkout this [github example](https://github.com/setiah/OpenSearchRestClient/tree/main) for full code.
+
+Congratulations, you have now successfully setup the Java high level REST client to connect with OpenSearch on a secure HTTPS channel. If you face any problems or would like to provide some feedback, please comment on this [github issue](https://github.com/opensearch-project/project-website/issues/440). 
```

Comment:
Right now the CTA is to send the users to the issue for this blog post. Can you provide a more meaningful CTA?

Reply:
That's a great point actually. Updated it in the latest version. Let me know if the new one works better.